testScripts/OfferingDownloadMedia.js

```diff
@@ -0,0 +1,339 @@
+/* eslint-disable no-console */
+const ScriptBase = require("./parentClasses/ScriptBase");
+const utils = require("../src/Utils");
+const Fraction = require("fraction.js");
+const {JSONPath} = require("jsonpath-plus");
+const fs = require("fs");
+const path = require("path");
+const { execSync } = require("child_process");
+
+class OfferingDownloadMedia extends ScriptBase {
+
+  async body(){
+
+    let objectId=this.args.objectId;
+    const versionHash=this.args.versionHash;
+    const offeringKey=this.args.offeringKey;
+    const streamKey=this.args.streamKey;
+    const startTime=this.args.startTime;
+    let endTime=this.args.endTime;
+    const out=this.args.out;
+
+    if(!objectId && !versionHash) throw new Error("Require object-id or object-hash to be provided");
+    if(!objectId) {
+      objectId = utils.DecodeVersionHash(versionHash).objectId;
+    }
+
+    const client=await this.client();
+    const libraryId=await client.ContentObjectLibraryId({objectId, versionHash});
+
+    const dirPath=path.resolve(out);
+    if(!fs.existsSync(dirPath)) {
+      fs.mkdirSync(dirPath, {recursive: true});
+      console.log(`Directory created at ${dirPath}`);
+    } else {
+      console.log(`Directory already exists at ${dirPath}`);
+    }
+
+    const contentObjDirPath = path.join(dirPath, `${objectId}_${this.secondsToHms(startTime, "-")}_${this.secondsToHms(endTime, "-")}`);
+    if(fs.existsSync(contentObjDirPath)) throw new Error(`Directory already exists at ${contentObjDirPath}`);
+    fs.mkdirSync(contentObjDirPath, { recursive: true });
+    console.log(`Directory created at ${contentObjDirPath}`);
+
+    // get object metadata
+    const metadata = await client.ContentObjectMetadata({ libraryId, objectId });
+
+    let audioFile = "", videoFile = "", audioVideoFile = "";
+    let minStart, maxEnd;
+    const processStream = async (key) => {
+      try {
+        const result = await this.processAudioOrVideoStream({
+          libraryId,
+          objectId,
+          metadata,
+          streamKey: key,
+          offeringKey,
+          startTime,
+          endTime,
+          out: contentObjDirPath,
+        });
+        console.log(`${key.toUpperCase()} MEDIA FILE:`, result.mediaFile);
+        return result;
+      } catch(error) {
+        throw new Error(`Error processing ${key}: ${error}`);
+      }
+    };
+
+    if(streamKey === "both") {
```

Comment:
The function works well. A few action items to generalize the method:
1. Support for Multiple Audio Streams
- Content object could have multiple audio streams, for instance `english_5_1, english_stereo, french_parisian_5_1, french_parisian_stereo`. 
-  Allow users to specify multiple stream names in the --streamKey argument, separated by spaces (e.g., --streamKey "video english_stereo french_parisian_stereo").
- Check if the specified streams exist in the content metadata before downloading.
2. Handling Duplicate Downloads
- When downloading the same content with the same start and end time but different streams, get an error `Directory already exists at /Users/jenniezhang/elvProjects/elv-client-js
Error: Directory already exists at /Users/jenniezhang/elvProjects/elv-client-js/iq__3QhxBgWHZDkcN87irAYqf5hSGdya_00-01-09.820_00-02-04.820`, the method should:
    - Avoid errors due to existing directories.
    - Allow downloading multiple streams for the same content without conflicts.

Example commands:
```
Download video: node ./testScripts/OfferingDownloadMedia.js  --startTime 69.82 --endTime 124.82 --objectId=iq__3QhxBgWHZDkcN87irAYqf5hSGdya  --config-url "https://main.net955305.contentfabric.io/config" --out . --streamKey video
```

```
Download audio: node ./testScripts/OfferingDownloadMedia.js  --startTime 69.82 --endTime 124.82 --objectId=iq__3QhxBgWHZDkcN87irAYqf5hSGdya  --config-url "https://main.net955305.contentfabric.io/config" --out . --streamKey "audio english_stereo french_parisian_stereo"
```